src/components/panel/panel.js

```diff
@@ -973,14 +977,24 @@ MdPanelRef.prototype.hide = function() {
   }
 
   var self = this;
+
   return this._$q(function(resolve, reject) {
     var done = self._done(resolve, self);
+    var focusOnOrigin = function (response) {
+      self.addClass(MD_PANEL_HIDDEN);
+      var origin = self._config['origin'];
+      if (origin) {
+        getElement(origin).focus();
+      }
+      return response;
+    };
 
     self._$q.all([
-      self._backdropRef ? self._backdropRef.hide() : self,
-      self._animateClose().then(function() { self.addClass(MD_PANEL_HIDDEN); },
-          reject)
-    ]).then(done, reject);
+        self._backdropRef ? self._backdropRef.hide() : self,
+        self._animateClose().then(focusOnOrigin)
```

Comment:
Can you rename your function? The most important part of this entire function is the hidden class.